### PR TITLE
Implement enum caching to speed tup object deletes

### PR DESCRIPTION
**Pull request by Apollon77**

### CODE REVIEW

File: packages/common/lib/common/tools.js [hunk 1]
```diff
@@ -2317,35 +2317,43 @@ function formatAliasValue(sourceObj, targetObj, state, logger, logNamespace) {
...
-                        }));
-                    } // endIf
-                } // endFor
-                Promise.all(promises).then(resolve);
-            } // endElse
         });
-    });
+        if (res && res.rows) {
+            for (let t = 0; t < res.rows.length; t++) {
+                allEnums[res.rows[t].id] = res.rows[t].value;
```
Comment (AlCalzone) [inline packages/common/lib/common/tools.js:2334]:
```suggestion
            for (const row of res.rows) {
                allEnums[row.id] = row.value;
```

File: packages/adapter/lib/adapter/adapter.js [hunk 2]
```diff
@@ -191,6 +189,7 @@ function Adapter(options) {
     this.logList = [];
     this.aliases = {};
     this.aliasPatterns = [];
+    this.enums = {};
```
Comment (AlCalzone) [inline packages/adapter/lib/adapter/adapter.js:192]: I'll advocate for using [Maps](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) over plain objects for lookup, but you decide.

File: packages/common/lib/common/tools.js [hunk 3]
```diff
@@ -2317,35 +2317,43 @@ function formatAliasValue(sourceObj, targetObj, state, logger, logNamespace) {
...
+    }
+
+    let error = null;
+    for (const enumId of Object.keys(allEnums)) {
+        const idx = allEnums[enumId].common.members ? allEnums[enumId].common.members.indexOf(id) : -1;
+        if (idx !== -1) {
+            // the id is in the enum now we have to remove it
+            allEnums[enumId].common.members.splice(idx, 1);
+            try {
+                await objects.setObjectAsync(enumId, allEnums[enumId]);
```
Comment (AlCalzone) [inline packages/common/lib/common/tools.js:2346]:
```suggestion
    for (const [enumId, enum] of Object.entries(allEnums)) {
        const idx = enum.common.members ? enum.common.members.indexOf(id) : -1;
        if (idx !== -1) {
            // the id is in the enum now we have to remove it
            enum.common.members.splice(idx, 1);
            try {
                await objects.setObjectAsync(enumId, enum);
```

**Review by AlCalzone — CHANGES_REQUESTED** [review]

File: packages/adapter/lib/adapter/adapter.js (hunk 2 above)
Comment (Apollon77) [reply packages/adapter/lib/adapter/adapter.js:192]: We have objects everywhere right now ... we can start mixing ... what do the overs think?

Comment (Apollon77) [reply packages/adapter/lib/adapter/adapter.js:192]: lets fdo that later. I create an issue

**Review by foxriver76 — APPROVED** [review]

**Pull request merged**